src/generators/BaseGenerator.js

```diff
@@ -0,0 +1,98 @@
+import fs from 'fs';
+import handlebars from 'handlebars';
+import mkdirp from 'mkdirp';
+import { sprintf } from 'sprintf-js';
+import urlapi from 'url';
+
+export default class {
+  templates = {};
+
+  constructor({hydraPrefix, templateDirectory}) {
+    this.hydraPrefix = hydraPrefix;
+    this.templateDirectory = templateDirectory;
+
+    this.registerTemplates('', ['_entrypoint.js']);
+  }
+
+  registerTemplates(basePath, paths) {
+    for (let path of paths) {
+      this.templates[path] = handlebars.compile(fs.readFileSync(`${this.templateDirectory}/${basePath}${path}`).toString());
```

Comment:
if only we could use `async / await` haha I don't like synchronous IO

Reply:
In this case (1-time file generation) synchronous IO is exactly what we want!